net/sockstats/sockstats_tsgo_darwin.go

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+//go:build tailscale_go && (darwin || ios)
+
+package sockstats
+
+import (
+	"syscall"
+
+	"golang.org/x/sys/unix"
+)
+
+func init() {
+	tcpConnStats = darwinTcpConnStats
+}
+
+func darwinTcpConnStats(c syscall.RawConn) (tx, rx uint64) {
+	c.Control(func(fd uintptr) {
```

Comment:
this might allocate the closure on the heap. but I guess that's ok because it's called once per connection?

Reply:
Fair, it might. We'll be allocating in the maps during that time too, so it probably falls out in the wash, but if we wanted to eradciate it completely we could capture our own fd table earlier in the process, or add a more direct API to get at the fd from a netconn - the rules for not breaking the reactor are straightforward so I'd be fine with us exposing the raw fd's if necessary.

Reply:
Since this is all for validation, I can make the entire code path opt-in, triggered by the `https://tailscale.com/cap/debug` node attr (or a more specific one)

Reply:
Actually, now that this is Darwin and TCP-only I think it's infrequent enough that it's not worth the extra knob.